src/ui/public/state_management/state_monitor.js

```diff
@@ -50,6 +55,17 @@ function stateMonitor(state, defaultState) {
   };
 
   return {
+    setDefault(defaultState) {
+      // update the originalState and apply ignoredProps
+      if (defaultState) {
+        setOriginalState(defaultState);
+        filterState(originalState);
```

Comment:
It gets used in `getStatus`, but you're right, most of the time it relies on object mutation.